app/src/main/java/ch/epfl/sdp/drone3d/drone/DroneServiceImpl.kt

```diff
@@ -11,30 +11,31 @@ package ch.epfl.sdp.drone3d.drone
 import io.mavsdk.System
 import io.mavsdk.mavsdkserver.MavsdkServer
 
-object DroneServiceImpl : DroneService {
+private const val DEFAULT_IP = "unknown"
+private const val DEFAULT_PORT = "-"
+
+class DroneServiceImpl(private val droneFactory: DroneServerFactory) : DroneService {
 
     private val droneData = DroneDataImpl(this)
 
-    private var isConnected = false
+    private var server: MavsdkServer? = null
+    private var droneInstance: System? = null
+
     private var isSimulation = true
-    private var simIP = "0.0.0.0"
-    private var simPort = "0"
-
-    override fun provideDrone(): System {
-        return if (isSimulation) {
-            val mavsdkServer = MavsdkServer()
-            val cloudSimIPAndPort = "tcp://$simIP:$simPort"
-            val mavsdkServerPort = mavsdkServer.run(cloudSimIPAndPort)
-            System("localhost", mavsdkServerPort)
-        } else {
-            val mavsdkServer = MavsdkServer()
-            val mavsdkServerPort = mavsdkServer.run()
-            System("localhost", mavsdkServerPort)
-        }
-    }
+    private var simIP = DEFAULT_IP
+    private var simPort = DEFAULT_IP
```

Comment:
Those 2 equalities seem odd as having the same value

Reply:
It was an error and has been corrected

Reply:
true, that's a bad copy/paste